### ISSUE TITLE
Missing execution of test scenarios in feature??

### ISSUE THREAD

**Issue by ranjith36** [prashant-ramcharan/courgette-jvm#7]
I've been using courgette-JVM in my project for a while, recently I've found out that it's sometimes missing execution of some test scenarios in the feature.

For Ex:
If my first test scenario failed in the feature, which contains 6 scenarios in total. courgette-jvm is immediately or trying to re-run failed test scenario instead of executing rest of the scenarios in the feature. by this way sometimes it's missing execution of some test scenarios in the feature.

**Comment by ranjith36** [comment]
Or It may be an issue with reports as well!! as it's executing failed test scenario at the end, so it might be overwriting the existing report that's why it's displaying only failed test scenario in report.

**Comment by ranjith36** [comment]
Forgot to mention I'm running at feature level!!

**Comment by prashant-ramcharan** [comment]
If you running on a feature level then the report and behaviour is correct!!

Example:

- You run 1 feature file (using feature level) with 6 scenarios.
- All 6 scenarios are executed on a single thread in consecutive order.
- The 3rd scenario fails.
- The 3rd scenario is re-run and the re-run fails.
- The remaining scenarios are executed as usual.

The expected result in the Courgette Report for that feature is a **failure**.

When running on a **feature level**, any single scenario failure will result in the overall feature been marked as a failure.

This does not mean the other scenarios did not execute and pass - it just means, all scenarios did not pass in the feature, so the entire feature is marked as a failure!

**Comment by ranjith36** [comment]
That's true!! but in the report, it should display all 6 scenarios including failed one but it only showing failed test scenario!!

**Comment by prashant-ramcharan** [comment]
Can you reproduce? If so, share the steps and the cucumber report!

I've not seen this issue with any of test runs thus far.

**Comment by prashant-ramcharan** [comment]
@ranjith36 will look into the code and fix if required.

Can you use scenario level for now?

**Comment by ranjith36** [comment]
Nope! because of the same issue.

I can reproduce this in `courgette-jvm-example` project by changing feature file `FeatureWithExamples.feature` to

@regression
Feature: Feature with Examples

  Scenario Outline: Ensure that stack overflow question pages can be opened
    When I navigate to Stack Overflow question page <page>
    Then I verify Stack Overflow question page <page2> is opened

    Examples:
      | page | page2 |
      | 3    |   3          |
      | 4    |   3          |
      | 5    |   5          |

then `gradle runFeatures` will give report

![screen shot 2017-08-21 at 17 13 26](https://user-images.githubusercontent.com/9530243/29528488-226baa50-8694-11e7-9762-159ea9972eef.png)

**Comment by prashant-ramcharan** [comment]
Will fix in next release. Thanks.

**Comment by prashant-ramcharan** [comment]
FYI: Using scenario level does give the correct report.

![image](https://user-images.githubusercontent.com/2563149/29531635-43b4cb6c-86a1-11e7-8039-be8e14313aa1.png)

Changing feature level reports would require extra work as the diff between the original run and re-run would need further parsing.

**Comment by prashant-ramcharan** [comment]
I've decided not to implement the diff check between original and re-run report when running on a **feature level**. 

There's no clean way of substituting data in the original html (+ json) after the re-run has complete as this Cucumber-JVM code not Courgette. 

I've released version 1.4.0 which includes a **Courgette-JVM html report**. This report is clear, searchable and paginated. [Screenshot here](https://postimg.org/image/m793ix42d/)

I would also suggest using SCENARIO run level.
